### PR TITLE
Added option to filter the session list insideTracksActivity.

### PR DESCRIPTION
**Pull request by Devastrix**
A filter button added in toolbar of TracksActivity.java to enable filter search. Resetting the filter option is also available.

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+@SuppressWarnings("deprecation")
+/**
+ * Created by user on 02-06-2015.
+ */
+public class FilterAdapter extends RecyclerView.Adapter<FilterAdapter.MyViewHolder> {
+    private LayoutInflater inflater;
```
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: You don't really require a different adapter for this. You can us the existing one.

File: app/src/main/java/org/fossasia/openevent/data/Information.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.fossasia.openevent.data;
+
+/**
+ * Created by user on 2/29/2016.
+ */
+public class Information {
+    public int itemId;
+    public String Title;
```
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/data/Information.java]: Why is this required? Better naming and explanation required

File: app/src/main/java/org/fossasia/openevent/utils/SaveFilters.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    public boolean saveArray(Boolean[] array, String arrayName, Context mContext) {
+        SharedPreferences prefs = mContext.getSharedPreferences("Filters", 0);
+        SharedPreferences.Editor editor = prefs.edit();
+        editor.putInt(arrayName +"_size", array.length);
+        for(int i=0;i<array.length;i++)
+            editor.putBoolean(arrayName + "_" + i, array[i]);
+
+        return editor.commit();
+    }
```
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/utils/SaveFilters.java]: Why would I want to save  the filtered array in SharedPreferences?

File: gradle.properties [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-# org.gradle.parallel=true
\ No newline at end of file
+# org.gradle.parallel=true
+#Tue Mar 01 21:36:15 IST 2016
+systemProp.http.proxyPassword=putu1
+systemProp.https.proxyUser=iit2013122
+systemProp.https.proxyPort=8080
+systemProp.http.proxyHost=172.31.1.4
+systemProp.https.proxyPassword=putu1
```
Comment (mananwason) [inline gradle.properties]: Please remove all these proxy settings

**Comment by mananwason** [comment]
@Devastrix Some fixes required.

File: app/src/main/java/org/fossasia/openevent/utils/SaveFilters.java (hunk 3 above)
Comment (Devastrix) [inline app/src/main/java/org/fossasia/openevent/utils/SaveFilters.java]: Suppose we are viewing the sessions of a particular track and we set the filters. Now we want to change alter the filters, so when we again open the filter dialog, the previous filters should be there. So we want to save the filters tiil we are in the Activity.

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java (hunk 1 above)
Comment (Devastrix) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Required because the recyclerview loads the data from sharedPreference.

File: app/src/main/java/org/fossasia/openevent/activities/TracksActivity.java [hunk 5]
```diff
@@ -44,6 +54,9 @@
     private String searchText = "";
 
     private SearchView searchView;
+    private RecyclerView recyclerView;
+    static String[] title = { "DevOps", "General", "OpenTech", "Web","Python", "Mozilla", "Exhibition"};
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/activities/TracksActivity.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406771674/issues/source?bid=3233850&fileBranchId=3233908#l58)

File: app/src/main/java/org/fossasia/openevent/activities/TracksActivity.java [hunk 6]
```diff
@@ -44,6 +54,9 @@
...
+    private TextView reset;
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/activities/TracksActivity.java]: Codacy Issue found: [Perhaps 'reset' could be replaced by a local variable.](https://www.codacy.com/app/triplez/open-event-android/file/2406771674/issues/source?bid=3233850&fileBranchId=3233908#l59)

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+          //  item.setTypeface(tf);
+
+            itemView.setOnClickListener(this);
+        }
+
+        @Override
+        public void onClick(View v) {
+            int itemPos = getPosition();
+            CheckBox cb = (CheckBox)v.findViewById(R.id.checkItem);
+            TextView f = (TextView)v.findViewById(R.id.itemTitle);
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Codacy Issue found: [Avoid unused local variables such as 'f'.](https://www.codacy.com/app/triplez/open-event-android/file/2406771676/issues/source?bid=3233850&fileBranchId=3233908#l91)

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+
+    @Override
+    public int getItemCount() {
+        return data.size();
+    }
+
+
+    class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        TextView item;
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406771676/issues/source?bid=3233850&fileBranchId=3233908#l69)

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        CheckBox itemCheck;
+        Context c;
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406771676/issues/source?bid=3233850&fileBranchId=3233908#l71)

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.Collections;
+import java.util.List;
+
+@SuppressWarnings("deprecation")
+/**
+ * Created by user on 02-06-2015.
+ */
+public class FilterAdapter extends RecyclerView.Adapter<FilterAdapter.MyViewHolder> {
+    private LayoutInflater inflater;
+    Context context;
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406771676/issues/source?bid=3233850&fileBranchId=3233908#l27)

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Override
+    public int getItemCount() {
+        return data.size();
+    }
+
+
+    class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        TextView item;
+        CheckBox itemCheck;
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406771676/issues/source?bid=3233850&fileBranchId=3233908#l70)

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.List;
+
+@SuppressWarnings("deprecation")
+/**
+ * Created by user on 02-06-2015.
+ */
+public class FilterAdapter extends RecyclerView.Adapter<FilterAdapter.MyViewHolder> {
+    private LayoutInflater inflater;
+    Context context;
+    List<String> data = Collections.emptyList();
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406771676/issues/source?bid=3233850&fileBranchId=3233908#l28)

File: app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+    Boolean[] checkedFilters;
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/adapters/FilterAdapter.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/triplez/open-event-android/file/2406771676/issues/source?bid=3233850&fileBranchId=3233908#l30)

File: app/src/main/java/org/fossasia/openevent/utils/SaveFilters.java [hunk 14]
```diff
@@ -0,0 +1,41 @@
+package org.fossasia.openevent.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Log;
```
Comment (juslee) [inline app/src/main/java/org/fossasia/openevent/utils/SaveFilters.java]: Codacy Issue found: [Avoid unused imports such as 'android.util.Log'](https://www.codacy.com/app/triplez/open-event-android/file/2406771741/issues/source?bid=3233850&fileBranchId=3233908#l5)

Comment (the-dagger) [inline app/src/main/java/org/fossasia/openevent/utils/SaveFilters.java]: @juslee Can i work on this? Seems like the author isn't much active.
Cheers.
